pkg/auth_proxy/openstack.go

```diff
@@ -50,35 +125,94 @@ func (oh *OpenStackAuthHandler) ServeHTTP(rw http.ResponseWriter, r *http.Reques
 			username := r.FormValue("username")
 			password := r.FormValue("password")
 
-			ok, err := oh.authenticate(username, password)
+			log.Logger.Debugf("%s is setting as '%s'", SessionUsername, username)
+
+			sess.Values[SessionUsername] = username
+
+			vclient, clientErr := client.NewVisualizationClient(visualizationEndpoint, http.Client{}, openstackToken)
+			if clientErr != nil {
+				log.Logger.Errorf("Error during getting visualization client: %s", clientErr)
+				return
+			}
+			// Get the user by its name if exists
+			userGet, err := vclient.GetUserName(username)
 			if err != nil {
-				log.Logger.Infof("User %s is not authenticated. Due to err: %s", username, err)
-				http.Redirect(rw, r, "/auth/openstack", http.StatusInternalServerError)
+				log.Logger.Errorf("Error during getting User by Name err: %s", err)
+				return
 			}
+			user := client.User{}
+			var orgName string
+			var role string
+			if user == userGet {
+				user.Name = username
+				user.Password = password
+				user.Login = username
+				//user.Email = username + "@test.com"
 
-			if !ok {
-				log.Logger.Infof("User %s is not authenticated", username)
-				http.Redirect(rw, r, "/auth/openstack", http.StatusForbidden)
-			}
+				// Create User
+				_, err = vclient.CreateUser(user)
```

Comment:
We shouldn't create any entities in visualization-api or Grafana here.
https://github.com/zaletniy/visualization-api/blob/auth_proxy/pkg/auth_proxy/visualization.go is the place for this.

The responsibility of OpenStackAuthHandler
- check user's credentials in openstack - login to keystone
- if passed. Create GafanaUpdateCommand and save in session